### PR TITLE
A ton of Quality of Life features 

### PR DESCRIPTION
**Pull request by Lucci228**

### CODE REVIEW

File: game/enemies/enemy.py [hunk 1]
```diff
@@ -14,7 +17,7 @@
 class Enemy(pygame.sprite.Sprite):
     def __init__(self, spritesheet, colisionHandler, wander_time: int, frame_width:int, frame_height:int,
                  num_frames, x, y, speed, attack_type, attack_damage, attack_range, health, colision_group,
-                 sprites_group, enemy_group, enemy_type='default', scale=1, player=None, projectile_path=None, projectile_cooldown=0, type="enemy"):
+                 sprites_group, enemy_group, enemy_type='default', scale=1, player : Player=None, projectile_path=None, projectile_cooldown=0, type="enemy"):
```
Comment (AndreiP256) [inline game/enemies/enemy.py:20]: If you are going to add types, add them to all parameters

File: game/groups/all_sprites_group.py [hunk 2]
```diff
@@ -13,17 +17,18 @@ def __init__(self):
...
-        ground_sprites = [sprite for sprite in self if hasattr(sprite, "ground")]
-        object_sprites = [sprite for sprite in self if not hasattr(sprite, "ground")]
-        for layer in [ground_sprites, object_sprites]:
-            for sprite in sorted(layer, key = lambda sprite: sprite.rect.centery):
-                self.surface.blit(sprite.image, sprite.rect.topleft + self.offset)
+        # ground_sprites = [sprite for sprite in self if hasattr(sprite, "ground")]
+        # object_sprites = [sprite for sprite in self if not hasattr(sprite, "ground")]
+        # for layer in [ground_sprites, object_sprites]:
+        #     for sprite in sorted(layer, key = lambda sprite: sprite.rect.centery):
+        #         self.surface.blit(sprite.image, sprite.rect.topleft + self.offset)
```
Comment (AndreiP256) [inline game/groups/all_sprites_group.py:27]: Why did you remove y ordering ?

File: game/player/Camera.py
Comment (AndreiP256) [inline game/player/Camera.py:1]: Why did you delete camera.py

File: game/groups/all_sprites_group.py (hunk 2 above)
Comment (Lucci228) [reply game/groups/all_sprites_group.py:27]: Because it looke bad

File: game/player/Camera.py
Comment (Lucci228) [reply game/player/Camera.py:1]: We didnt use it

File: game/enemies/enemy.py (hunk 1 above)
Comment (Lucci228) [reply game/enemies/enemy.py]: I fixed it by removing the type for player

**Review by AndreiP256 — APPROVED** [review]
Ok-ish work,not 100% satisfied but ok overall
